src/core/ddsi/src/q_ddsi_discovery.c

```diff
@@ -1161,8 +1161,8 @@ static int sedp_write_endpoint_impl
       addrset_forall (as, add_xlocator_to_ps, &arg);
 
 #ifdef DDS_HAS_SHM
-    if ((xqos->present & QP_SHARED_MEMORY) == QP_SHARED_MEMORY &&
-      xqos->shared_memory.enabled)
+    if (xqos->present & QP_LOCATOR_MASK &&
```

Comment:
Same here, no need for the `xqos->present & QP_LOCATOR_MASK` anymore

Reply:
Indeed.